app/experimenter/nimbus-ui/src/components/PageRequestReview/index.stories.tsx

```diff
@@ -3,134 +3,89 @@
+  mock,
+  reviewPendingBaseProps,
+  reviewRejectedBaseProps,
+  reviewRequestedBaseProps,
+  reviewTimedoutBaseProps,
+  Subject,
+} from "./mocks";
 
-storiesOf("pages/RequestReview", module)
-  .addDecorator(withLinks)
-  .add("draft status", () => {
-    const { mock } = mockExperimentQuery("demo-slug", {
-      status: NimbusExperimentStatus.DRAFT,
-    });
-    return <Subject {...{ mocks: [mock] }} />;
-  })
-  .add("preview status", () => {
-    const { mock } = mockExperimentQuery("demo-slug", {
-      status: NimbusExperimentStatus.PREVIEW,
-    });
-    return <Subject {...{ mocks: [mock] }} />;
-  })
-  .add("preview status + approved publish status", () => {
-    const { mock } = mockExperimentQuery("demo-slug", {
-      status: NimbusExperimentStatus.PREVIEW,
-      publishStatus: NimbusExperimentPublishStatus.APPROVED,
-    });
-    return <Subject {...{ mocks: [mock] }} />;
-  })
-  .add("preview status + publish status waiting", () => {
-    const { mock } = mockExperimentQuery("demo-slug", {
-      status: NimbusExperimentStatus.PREVIEW,
-      publishStatus: NimbusExperimentPublishStatus.WAITING,
-    });
+const storyWithExperimentProps = (
+  props: Partial<getExperiment_experimentBySlug | null>,
+  storyName?: string,
+) => {
+  const story = () => {
+    const { mock } = mockExperimentQuery("demo-slug", props);
     return <Subject {...{ mocks: [mock] }} />;
-  })
-  .add("review status", () => {
-    const { mock } = mockExperimentQuery("demo-slug", {
-      publishStatus: NimbusExperimentPublishStatus.REVIEW,
-    });
-    return <Subject {...{ mocks: [mock] }} />;
-  })
-  .add("non-reviewable", () => {
-    const { mock } = mockExperimentQuery("demo-slug", {
-      publishStatus: NimbusExperimentPublishStatus.WAITING,
-    });
-    return <Subject {...{ mocks: [mock] }} />;
-  })
-  .add("error", () => (
-    <RouterSlugProvider mocks={[mock]}>
-      <PageRequestReview polling={false} />
-    </RouterSlugProvider>
-  ));
+  };
+  story.storyName = storyName;
+  return story;
+};
+
+export default {
+  title: "pages/RequestReview",
+  component: Subject,
+  decorators: [withLinks],
+};
+
+export const draftStatus = storyWithExperimentProps({
+  status: NimbusExperimentStatus.DRAFT,
+  publishStatus: NimbusExperimentPublishStatus.IDLE,
+});
+
+export const previewStatus = storyWithExperimentProps({
+  status: NimbusExperimentStatus.PREVIEW,
+  publishStatus: NimbusExperimentPublishStatus.IDLE,
+});
+
+export const reviewRequestedCanReview = storyWithExperimentProps(
+  { ...reviewRequestedBaseProps, canReview: true },
+  "Review requested, user can review",
+);
+
+export const reviewPendingCanReview = storyWithExperimentProps(
+  { ...reviewPendingBaseProps, canReview: true },
+  "Review pending in Remote Rettings, user can review",
+);
+
+export const reviewTimedoutCanReview = storyWithExperimentProps(
+  { ...reviewTimedoutBaseProps, canReview: true },
+  "Review timed out in Remote Settings, user can review",
```

Comment:
I think this is a bug in the timeout celery task - it rolls back to IDLE rather than REVIEW publishStatus. I've got a PR here:

https://github.com/mozilla/experimenter/pull/5035

Both the UI component and the `latest_timeout` model method are looking for a transition from WAITING to REVIEW. Transitioning from WAITING to IDLE looks like a rejection.

Reply:
Just rolled PR #5035 into this PR.